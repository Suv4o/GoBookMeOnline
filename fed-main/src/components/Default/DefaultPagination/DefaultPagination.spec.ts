import { render, cleanup } from '@testing-library/vue'
import { describe, it, expect } from 'vitest'
import DefaultPagination from './DefaultPagination.vue'

describe('DefaultPagination', () => {
  it('render component correctly', async () => {
    render(DefaultPagination)
    cleanup()
  })

  it('snap shot matches', async () => {
    const wrapper = render(DefaultPagination)
    expect(wrapper).toMatchSnapshot()
  })
})
