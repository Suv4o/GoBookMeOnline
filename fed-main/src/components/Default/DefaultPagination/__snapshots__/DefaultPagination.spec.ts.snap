// Vitest Snapshot v1

exports[`DefaultPagination > snap shot matches 1`] = `
{
  "baseElement": <body>
    <div>
      <div
        class="default-container"
      >
        <nav
          class="border-t border-gray-200 px-4 flex items-center justify-between sm:px-0"
        >
          <div
            class="-mt-px w-0 flex-1 flex"
          >
            <a
              class="border-t-2 border-transparent pt-4 pr-1 inline-flex items-center text-sm font-medium text-gray-500 hover:text-gray-700 hover:border-gray-300"
              href="#"
            >
              <svg
                aria-hidden="true"
                class="mr-3 h-5 w-5 text-gray-400"
                fill="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M7.28 7.72a.75.75 0 010 1.06l-2.47 2.47H21a.75.75 0 010 1.5H4.81l2.47 2.47a.75.75 0 11-1.06 1.06l-3.75-3.75a.75.75 0 010-1.06l3.75-3.75a.75.75 0 011.06 0z"
                  fill-rule="evenodd"
                />
              </svg>
               Previous 
            </a>
          </div>
          <div
            class="hidden md:-mt-px md:flex"
          >
            <a
              class="border-teal-500 text-teal-600 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium"
              href="#"
            >
               1 
            </a>
            <a
              aria-current="page"
              class="border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium"
              href="#"
            >
               2 
            </a>
            <a
              class="border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium"
              href="#"
            >
               3 
            </a>
            <span
              class="border-transparent text-gray-500 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium"
            >
               ... 
            </span>
            <a
              class="border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium"
              href="#"
            >
               8 
            </a>
            <a
              class="border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium"
              href="#"
            >
               9 
            </a>
            <a
              class="border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium"
              href="#"
            >
               10 
            </a>
          </div>
          <div
            class="-mt-px w-0 flex-1 flex justify-end"
          >
            <a
              class="border-t-2 border-transparent pt-4 pl-1 inline-flex items-center text-sm font-medium text-gray-500 hover:text-gray-700 hover:border-gray-300"
              href="#"
            >
               Next 
              <svg
                aria-hidden="true"
                class="ml-3 h-5 w-5 text-gray-400"
                fill="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M16.72 7.72a.75.75 0 011.06 0l3.75 3.75a.75.75 0 010 1.06l-3.75 3.75a.75.75 0 11-1.06-1.06l2.47-2.47H3a.75.75 0 010-1.5h16.19l-2.47-2.47a.75.75 0 010-1.06z"
                  fill-rule="evenodd"
                />
              </svg>
            </a>
          </div>
        </nav>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="default-container"
    >
      <nav
        class="border-t border-gray-200 px-4 flex items-center justify-between sm:px-0"
      >
        <div
          class="-mt-px w-0 flex-1 flex"
        >
          <a
            class="border-t-2 border-transparent pt-4 pr-1 inline-flex items-center text-sm font-medium text-gray-500 hover:text-gray-700 hover:border-gray-300"
            href="#"
          >
            <svg
              aria-hidden="true"
              class="mr-3 h-5 w-5 text-gray-400"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M7.28 7.72a.75.75 0 010 1.06l-2.47 2.47H21a.75.75 0 010 1.5H4.81l2.47 2.47a.75.75 0 11-1.06 1.06l-3.75-3.75a.75.75 0 010-1.06l3.75-3.75a.75.75 0 011.06 0z"
                fill-rule="evenodd"
              />
            </svg>
             Previous 
          </a>
        </div>
        <div
          class="hidden md:-mt-px md:flex"
        >
          <a
            class="border-teal-500 text-teal-600 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium"
            href="#"
          >
             1 
          </a>
          <a
            aria-current="page"
            class="border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium"
            href="#"
          >
             2 
          </a>
          <a
            class="border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium"
            href="#"
          >
             3 
          </a>
          <span
            class="border-transparent text-gray-500 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium"
          >
             ... 
          </span>
          <a
            class="border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium"
            href="#"
          >
             8 
          </a>
          <a
            class="border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium"
            href="#"
          >
             9 
          </a>
          <a
            class="border-transparent text-gray-500 hover:text-gray-700 hover:border-gray-300 border-t-2 pt-4 px-4 inline-flex items-center text-sm font-medium"
            href="#"
          >
             10 
          </a>
        </div>
        <div
          class="-mt-px w-0 flex-1 flex justify-end"
        >
          <a
            class="border-t-2 border-transparent pt-4 pl-1 inline-flex items-center text-sm font-medium text-gray-500 hover:text-gray-700 hover:border-gray-300"
            href="#"
          >
             Next 
            <svg
              aria-hidden="true"
              class="ml-3 h-5 w-5 text-gray-400"
              fill="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M16.72 7.72a.75.75 0 011.06 0l3.75 3.75a.75.75 0 010 1.06l-3.75 3.75a.75.75 0 11-1.06-1.06l2.47-2.47H3a.75.75 0 010-1.5h16.19l-2.47-2.47a.75.75 0 010-1.06z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </div>
      </nav>
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
